### PR TITLE
Fixes #329. Multiple Fixes and Improvements as stated in #330

### LINKED ISSUE #329: (simpleapp) App Crashes when user click's Log Out.
Logcat and fix for this issue attached:

Logcat: `Process: dev.shreyaspatil.noty.simpleapp, PID: 14678
    java.lang.IllegalArgumentException: Navigation action/destination dev.shreyaspatil.noty.simpleapp:id/action_notesFragment_to_loginFragment cannot be found from the current destination Destination(dev.shreyaspatil.noty.simpleapp:id/loginFragment) label=fragment_login class=dev.shreyaspatil.noty.simpleapp.view.login.LoginFragment
        at androidx.navigation.NavController.navigate(NavController.java:940)`

Fix: To check current destination before navigating.

I can contribute to this issue as well if @PatilShreyas assigns me 😅.

### PR DESCRIPTION
**Pull request by kasem-sm**
## Description for the changelog
Stated [here ](https://github.com/PatilShreyas/NotyKT/discussions/330)in detail.

## Checklist

- [x] Build and linting is passing.
- [x] This change is not breaking existing flow of a system.
- [ ] I have written test case for this change.
- [x] This change is tested from all aspects.
- [ ] Implemented any new third-party library _(Which not existed before this change)_.

### CODE REVIEW

File: noty-android/app/build.gradle [hunk 1]
```diff
@@ -77,6 +77,10 @@ dependencies {
     // JetPack Security
     implementation "androidx.security:security-crypto:$securityCryptoVersion"
 
+    // Navigation
+    implementation "androidx.navigation:navigation-fragment-ktx:$navigationVersion"
+    implementation "androidx.navigation:navigation-ui-ktx:$navigationVersion"
+
```
Comment (PatilShreyas) [inline noty-android/app/build.gradle:83]: What's the need of including this in `app` module? It's already included in `simpleapp` module I guess.

File: noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/component/text/NoteTextFields.kt [hunk 2]
```diff
@@ -36,7 +36,7 @@ fun NoteTitleField(
         value = value,
         label = "Title",
         onTextChange = onTextChange,
-        textStyle = MaterialTheme.typography.h6,
+        textStyle = MaterialTheme.typography.h4.copy(fontWeight = FontWeight.Bold),
```
Comment (PatilShreyas) [inline noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/component/text/NoteTextFields.kt:39]: I guess `h6` was good. Doesn't it (title) look bigger than it should be?

File: noty-android/app/src/main/java/dev/shreyaspatil/noty/utils/ext/DialogExt.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+data class DialogComponents(
+    val title: String? = null,
+    val message: String? = null,
+    val positiveActionText: String? = null,
+    val positiveAction: (d: DialogInterface, i: Int) -> Unit = { _, _ -> },
+    val negativeActionText: String? = null,
+    val negativeAction: (d: DialogInterface, i: Int) -> Unit = { _, _ -> },
+    val isDismissAble: Boolean = true
+)
```
Comment (PatilShreyas) [inline noty-android/app/src/main/java/dev/shreyaspatil/noty/utils/ext/DialogExt.kt:32]: No need to create separate class for this, directly include these as params in `showDialog()` function

File: noty-android/app/src/main/java/dev/shreyaspatil/noty/utils/ext/NavigationExt.kt [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+import androidx.fragment.app.Fragment
+import androidx.navigation.NavDirections
+import androidx.navigation.findNavController
+
+fun Fragment.navigate(
+    destination: NavDirections
+) {
+    if (isAdded) {
+        with(view?.findNavController()) {
```
Comment (PatilShreyas) [inline noty-android/app/src/main/java/dev/shreyaspatil/noty/utils/ext/NavigationExt.kt:27]: we can directly get it using `Fragment` instance as
```suggestion
        with(findNavController()) {
```

File: noty-android/app/src/main/java/dev/shreyaspatil/noty/utils/ext/NavigationExt.kt [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+ * limitations under the License.
+ */
+
+package dev.shreyaspatil.noty.utils.ext
+
+import androidx.fragment.app.Fragment
+import androidx.navigation.NavDirections
+import androidx.navigation.findNavController
+
+fun Fragment.navigate(
```
Comment (PatilShreyas) [inline noty-android/app/src/main/java/dev/shreyaspatil/noty/utils/ext/NavigationExt.kt:23]: Also, I feel there's no special need for this function.

**Review by PatilShreyas — CHANGES_REQUESTED** [review]
@kasem-sm thanks for the PR. It's very neat and clean 😃. 
Did the first round of reviews and here are a few suggestions. I Will review again after trying a build.

File: noty-android/app/build.gradle (hunk 1 above)
Comment (PatilShreyas) [reply noty-android/app/build.gradle:83]: See this: https://github.com/PatilShreyas/NotyKT/blob/master/noty-android/app/simpleapp/build.gradle#L125-L126

Comment (kasem-sm) [reply noty-android/app/build.gradle:83]: oops will remove this from app module and migrate the extensions function to simpleapp module!

File: noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/component/text/NoteTextFields.kt (hunk 2 above)
Comment (kasem-sm) [reply noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/component/text/NoteTextFields.kt:39]: hmm yes 🤔

File: noty-android/app/src/main/java/dev/shreyaspatil/noty/utils/ext/DialogExt.kt (hunk 3 above)
Comment (kasem-sm) [reply noty-android/app/src/main/java/dev/shreyaspatil/noty/utils/ext/DialogExt.kt:32]: will do!

File: noty-android/app/src/main/java/dev/shreyaspatil/noty/utils/ext/NavigationExt.kt (hunk 5 above)
Comment (kasem-sm) [reply noty-android/app/src/main/java/dev/shreyaspatil/noty/utils/ext/NavigationExt.kt:23]: Yeah, currently as this function is used at only single place, will consider removing it.

File: noty-android/app/src/main/java/dev/shreyaspatil/noty/utils/ext/DialogExt.kt [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+
+fun Fragment.showDialog(
+    title: String? = null,
+    message: String? = null,
+    positiveActionText: String? = null,
+    positiveAction: (d: DialogInterface, i: Int) -> Unit = { _, _ -> },
+    negativeActionText: String? = null,
+    negativeAction: (d: DialogInterface, i: Int) -> Unit = { _, _ -> },
+    isDismissAble: Boolean = true
```
Comment (PatilShreyas) [inline noty-android/app/src/main/java/dev/shreyaspatil/noty/utils/ext/DialogExt.kt:31]: Rename this to `isCancelable`

**Review by PatilShreyas — CHANGES_REQUESTED** [review]

File: noty-android/app/src/main/java/dev/shreyaspatil/noty/utils/ext/DialogExt.kt (hunk 6 above)
Comment (kasem-sm) [reply noty-android/app/src/main/java/dev/shreyaspatil/noty/utils/ext/DialogExt.kt:31]: sure

**Review by PatilShreyas — APPROVED** [review]
Did final round of review and I would say, it's awesome 👍. Thank you @kasem-sm for such wonderful PR and it's surely helpful for improving quality of the application. Feel free to contribute again 😉

**Pull request merged**